### PR TITLE
migrate majority of tests to junit5

### PR DESCRIPTION
**Pull request by Siedlerchr**
please check if I didn't forget one Test annotation

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [ ] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/test/java/org/jabref/TestIconsProperties.java [hunk 1]
```diff
@@ -31,15 +31,14 @@ public void testExistenceOfIconImagesReferencedFromIconsProperties() throws IOEx
         try (Reader reader = Files.newBufferedReader(Paths.get(iconsPropertiesPath))) {
             properties.load(reader);
         }
-        assertFalse("There must be loaded properties after loading " + iconsPropertiesPath,
-                properties.entrySet().isEmpty());
+        assertFalse(properties.entrySet().isEmpty(), () -> "There must be loaded properties after loading " + iconsPropertiesPath);
```
Comment (stefan-kolb) [inline src/test/java/org/jabref/TestIconsProperties.java]: Hm, not sure if I like these Supplier Lambdas...Are they really needed? Makes the methods look more complex and I doubt that the evaluation of the message really takes a long time. WDYT?

Comment (Siedlerchr) [reply src/test/java/org/jabref/TestIconsProperties.java]: In the beginning I did not see that I can pass the message as parameter without a supplier . The later tests are without. The supplier is only needed in the assertThroes

**Comment by Siedlerchr** [comment]
I am not sure If I should merge this or wait for #3704

**Comment by koppor** [comment]
I am still struggling with Install4J. Not sure, when I'll have one that fight.

**Pull request merged**